### PR TITLE
[4.0] Fix missing onchange events on switcher fields

### PR DESCRIPTION
**Pull request by richard67**
Pull Request for Issue #20853 .

### Summary of Changes

Make it possible to use "onchange" events for switcher fields (type="radio", class="switcher").

There was once opened issue #20853 for "onchange" and "onclick" events, which was closed because PR #20931 was made for the CMS.

This PR then was closed because a PR against the backend template repo was made: [joomla/40-backend-template#446](https://github.com/joomla/40-backend-template/pull/446).

This then was merged into the backend template but either never found the way to the CMS or was later removed by some other change.

This PR here brings the correction back, but only for "onchange", not for "onclick", because to click on a switcher always means to change the value, and because we also don't have the "onclick" event on list fields.

The change does it in a bit different way than #20931 and [joomla/40-backend-template#446](https://github.com/joomla/40-backend-template/pull/446), so it is closer to the existing implementation in other field layouts.

In addition to the "onchange" property, this PR also adds the "id" property.

### Testing Instructions

1. Make sure that your web browser allows popup alerts for your Joomla backend.

2. Edit some form XML and add an onchange event to a switcher field.
E.g. you can edit file "administrator/components/com_config/forms/application.xml" and modify field name="gzip" by adding an "onchange" property for a javascript alert as follows:
```
<field
	name="gzip"
	type="radio"
	label="COM_CONFIG_FIELD_GZIP_COMPRESSION_LABEL"
	class="switcher"
	default="0"
	filter="boolean"
	onchange="alert('Hello, I am a switcher field.')"
	>
	<option value="0">JNO</option>
	<option value="1">JYES</option>
</field>
```

3. Go to a page where you can see the switcher field modified in step 2 and toggle the field.
E.g. if you used the field suggested in step 2: Go to the "Server" tab of "Global Configuration" in backend and toggle the "Gzip Page Compression" switcher field.
Result: No browser alert, see section "Actual result" below..

4. In developer tools of your web browser, inspect html of the switcher element ("fieldset" html element).
Result: No "onchange" property added to fieldset, see section "Actual result" below.

4. Apply the patch of this PR.

5. Go to some other page and then back, e.g. if you used "Global Configuration" close it and then open it again, and toggle the switcher field.
Result: See section "Expected result" below.

6. In developer tools of your web browser, inspect html of the switcher element ("fieldset" html element).
Result: See section "Expected result" below.

### Expected result

Onchange event works:

switcher-3

Fieldset has "id" and "onchange" properties:

switcher-2

### Actual result

Onchange event doesn't work.

Fieldset doesn't have "id" and "onchange" properties:

switcher-1

### Documentation Changes Required

Not sure. Maybe some doc about fields tells onchange is not supported for switchers?

### CODE REVIEW

**Comment by richard67** [comment]
Drone failure seems to be the usual, randomly occurring timeout on mysql system tests and so not to be related to this PR.

**Comment by richard67** [comment]
@brianteeman I'd be happy if you could find some time for testing this. It's an easy test for a small change.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 765163b60445e304336b6a5be27a09f8106a1e1b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27355">issues.joomla.org/tracker/joomla-cms/27355</a>.</sub>

File: layouts/joomla/form/field/radio/switcher.php [hunk 1]
```diff
@@ -60,8 +60,17 @@
  */
 $input    = '<input type="radio" id="%1$s" name="%2$s" value="%3$s" %4$s>';
 
+$attr = 'id="' . $id . '"';
+$attr .= $onchange ? ' onchange="' . $onchange . '"' : '';
+
+// To avoid user's confusion, readonly="readonly" should imply disabled="disabled".
+if ($readonly || $disabled)
+{
+	$attr .= ' disabled="disabled"';
```
Comment (SharkyKZ) [inline layouts/joomla/form/field/radio/switcher.php]: When `disabled` attribute is used, no values are sent to the server.

Comment (richard67) [reply layouts/joomla/form/field/radio/switcher.php]: Yes, and? Other fields use same or similar code.

Comment (richard67) [reply layouts/joomla/form/field/radio/switcher.php]: @SharkyKZ Check my previous comment and if there is still an issue be more precise what to change in  my code so I know what to do.

**Comment by richard67** [comment]
@SharkyKZ I've removed disabled and readonly stuff so this PR does not more than the description says. Someone else may later add back these in the right way and maybe add more properties if necessary with a new PR.

@Quy Could you test again?

**Comment by HLeithner** [comment]
Quy test is valid and merge by code review

**Comment by HLeithner** [comment]
Thanks

**Comment by richard67** [comment]
Thanks too to testers and mergers ;-)

**Pull request merged**